src/events/handlers.ts

```diff
@@ -0,0 +1,35 @@
+import { transactionsChannel } from './channels/transactionsChannel.js'
+import { logger } from '../modules/logger.js'
+import { processSignalTransaction } from '../services/processSignalTransaction.js'
+import { processTransactionToNotify } from '../services/processTransactionToNotify.js'
+import { prisma } from '../modules/prisma.js'
+
+export const spawnEventHandlers = () => {
+  transactionsChannel.on('newSignalMessage', (tx) => {
+    processSignalTransaction(tx).catch((e) =>
+      logger.error(e, 'Failed to process signal transaction')
+    )
+  })
+
+  transactionsChannel.on('newMessage', async (tx) => {
+    let devices = await prisma.device.findMany({
+      where: { admAddress: tx.recipientId }
+    })
+
+    devices = devices.filter((device) => device.admAddress === tx.recipientId)
+
+    if (!devices.length) {
+      return
+    }
+
+    logger.info(
+      `Got transaction to notify devices, devices ids: ${devices.map((d) => d.id).join(', ')}, providers: ${devices.map((d) => d.pushServiceProvider).join(', ')}, admTxId: ${tx.id}`
+    )
+
+    await Promise.all(
+      devices.map((device) => {
+        processTransactionToNotify(tx, device)
+      })
+    )
+  })
```

Comment:
Try to use [cargoQueue](https://caolan.github.io/async/v3/docs.html#cargoQueue) from `async` lib. I see that FCM supports notification batching. This should improve performance when sending a large number of notifications.

![image](https://github.com/user-attachments/assets/a2300142-915d-45cd-a414-970d4fac35e1)